week3/homework/src/index.js

```diff
@@ -1,84 +1,100 @@
 // 'use strict';
 
 const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
+
 const app = express();
-const validateTodos = require('./validateTodos');
-const todo = require('./todo.json');
 app.use(express.json());
 
 // get the list of todo
 app.get('/todo', (req, res) => {
-  res.send(todo);
+  fs.readFile('./todo.json', 'utf8', (error, data) => {
+    if (error) return res.status(500).send('no data found');
+    const todo = JSON.stringify(data);
+    res.json(todo);
+  });
 });
 
 // get todo by id number
 app.get('/todo/:id', (req, res) => {
-  const todoItem = todo.find(i => i.id === parseInt(req.params.id));
-  if (!todoItem) return res.status(404).send('Id not found');
-  res.send(todoItem);
+  fs.readFile('./todo.json', 'utf8', (error, data) => {
+    if (error) return res.status(400).send(`No data with id : ${id} found !`);
+    const id = req.param.id - 1;
+    const todo = JSON.stringify(data)[id];
+    res.json(todo);
+  });
 });
 
-// add todo
+// add todo item
 app.post('/todo', (req, res) => {
-  const { error } = validateTodos(req.body);
-  if (error) {
-    return res.status(400).send(error.details[0].message);
-  }
-
-  const todoItem = {
-    id: todo.length + 1,
-    name: req.body.description,
-    done: false,
-  };
-  todo.push(todoItem);
-  res.send(todo);
-});
-
-// modify todo
-app.put('/todo/:id', (req, res) => {
-  const todoItem = todo.find(i => i.id === parseInt(req.params.id));
-  if (!todoItem) return res.status(404).send('Id not found');
-
-  const { error } = validateTodos(req.body);
-  if (error) return res.status(400).send(error.details[0].message);
-
-  todoItem.description = req.body.description;
-  res.send(todo);
+  fs.readFile('./todo.json', 'utf8', (error, contents) => {
+    if (error) {
+      res.status(500).send('There is an error');
+    } else {
+      const todo = JSON.parse(contents);
+      const todoItem = {
+        id: todo.length + 1,
+        description: req.body.description,
+        done: false,
+      };
+      todo.push(todoItem);
+      const newTodo = JSON.stringify(todo);
+      fs.writeFile('/todo.json', newTodo, error => {
```

Comment:
exactly like this 👍

Reply:
Thanks a lot @gajduk ,  I learn alot during your review.